rust/gui-client/src-tauri/src/gui/system_tray.rs

```diff
@@ -324,11 +334,18 @@ impl SignedIn {
 
         if res.is_internet_resource() {
             submenu.add_separator();
-            if self.is_internet_resource_enabled() {
-                submenu.add_item(item(Event::DisableInternetResource, DISABLE));
-            } else {
-                submenu.add_item(item(Event::EnableInternetResource, ENABLE));
-            }
+
+            submenu = match (
+                self.is_internet_resource_enabled(),
+                self.internet_resource_force_enabled,
+            ) {
+                (true, None) => submenu.item(Event::DisableInternetResource, DISABLE),
+                (false, None) => submenu.item(Event::EnableInternetResource, ENABLE),
+
+                // If Internet Resource is MDM-controlled, don't let the user interact with it.
+                (_, Some(true)) => submenu.disabled(DISABLE),
+                (_, Some(false)) => submenu.disabled(ENABLE),
+            };
```

Comment:
@jamilbk What does the MacOS client do if the Internet Resource is force _disabled_? i.e. the policy is set but the boolean is false? I've opted to handle this as it is disabled and the user cannot enable it.